### PR TITLE
[SEMI-MODULAR] More lewd chemistry additions, plus extra misc text features. Adds OD to camphor, updates OD effects.

### PR DESCRIPTION
**Pull request by Gofawful5**
## About The Pull Request

Edit to succubus milk / incubus draft code that adds different text upon expansion / enlargement- this includes varying text for if the character's bust/penis is visible or not, and if the boobs are really big. The exact cup size change isn't visible to other players, but it is visible to the player who's being affected.

Adds overdose to camphor at 25 units, which allows for players to reset their genital sizes based on their preferences, and also changes gender if they have forced gender change enabled.

Adds breast/penis organs for ODing on the succubus/incubus chemicals, adjusted properly to fit lizard cocks as well.

Edits overdose effects to be minor / not conflicting.

## How This Contributes To The Skyrat Roleplay Experience

Adds more interaction for scenes, some immersion, and also a way of telling if the chemical is actually working.
Also adds more genderswap related stuff.
Finally, it adds a way of resetting genital sizes after scenes.

## Changelog

:cl: Goffy
add: Adds misc text for the growth effect of the succubus/incubus chemicals that will show up for other players/the player themselves, with different text shown publicly depending on if those genitals are clothed or not.
add: Adds functionality for genderswapping via ODing on such chemicals.
add: Adds OD to camphor which resets genital sizes based off of preferences.
/:cl:

### CODE REVIEW

**Comment by Zonespace27** [comment]
expansion tag's not used, i'd recommend the `Addition` tag over it.

**Comment by Gofawful5** [comment]
> expansion tag's not used, i'd recommend the `Addition` tag over it.

got it, done

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: Auto-doc and use `snake_case`

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: use early returns for a lot of this

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: Why have you got rid of the cap, exactly?

**Review by tf-4 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: I removed the cap because "beyond measurement" is never used, though it can be readded if it causes issues

Comment (tf-4) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: Unless I'm being blind, won't it just make them expand indefinitely?

Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: as long as the chemical's in them and continues increasing the enlargement_amount value, yeah. i don't believe it'd cause any issues, aside from it being significantly harder to make them smaller

Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: i'll readd a hard cap at the max breast size + 5, just so the beyond measurement text can be used - since it's unused in genitals.dm

**Comment by Gofawful5** [comment]
added a fix to continue dealing damage when clothes are being worn without the bust being exposed - at stupidly huge sizes

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (tf-4) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: I mean, I feel like `max_breast_size` should be... the maximum breast size. Increase the cap if you really want, I suppose.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]:
```suggestion
		/// Words for the breasts being huge.
```
Triple forward slash for auto-doc

**Comment by gonenoculer5** [comment]
Phenomenally based change. Succubus milk public play do be existing now.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: not like this. Increase the var directly. Should also probably be a define but ehhh.

Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: got it

**Comment by Gofawful5** [comment]
wait gdi

**Comment by Gofawful5** [comment]
iiii think i can add the effects that were planned but never added as well, ie adding breasts on male/breastless characters on OD, though i'll wait until this is reviewed before i commit those.
and i can do the same type of to_chat messages with cocks as well, but i think i'd make that a bit more subtle, since breasts are quite large compared to cocks. until they're huge. then i'unno

**Comment by gonenoculer5** [comment]
@Gofawful5 Does this include sprites for sizes 17-21?

**Comment by tf-4** [comment]
Actually to be honest, I'm not sure why the cap needs to be increased anyway? There are already plenty of sizes.

**Comment by Gofawful5** [comment]
> @Gofawful5 Does this include sprites for sizes 17-21?

no, but that's because i'd have to mess with genitals.dm
i have the sprites already, up to 25, and could add them in

**Comment by Gofawful5** [comment]
> Actually to be honest, I'm not sure why the cap needs to be increased anyway? There are already plenty of sizes.

oh, to use the text that appears when they're "beyond measurement." I could include sprites if needed, though I would have to change genitals.dm and do some math so they're not included in the player breast selection list.

**Comment by gonenoculer5** [comment]
> > Actually to be honest, I'm not sure why the cap needs to be increased anyway? There are already plenty of sizes.
> 
> oh, to use the text that appears when they're "beyond measurement." I could include sprites if needed, though I would have to change genitals.dm and do some math so they're not included in the player breast selection list.

Honestly more sprites and such would be based as fuck. Hyper has sprites if you need them ported but y'could always ask Arcie.

**Comment by Gofawful5** [comment]
> > > Actually to be honest, I'm not sure why the cap needs to be increased anyway? There are already plenty of sizes.
> > 
> > 
> > oh, to use the text that appears when they're "beyond measurement." I could include sprites if needed, though I would have to change genitals.dm and do some math so they're not included in the player breast selection list.
> 
> Honestly more sprites and such would be based as fuck. Hyper has sprites if you need them ported but y'could always ask Arcie.

the style of hyper's breast sprites are different. i'll commit the new sprites soon + fix the PR title and description with previews, but i'm trying to stick with the current style in order for it not to feel out of place
I'm also adding functionality for the ghetto gender swapping that was commented in but was never actually added

**Comment by gonenoculer5** [comment]
> > > > Actually to be honest, I'm not sure why the cap needs to be increased anyway? There are already plenty of sizes.
> > > 
> > > 
> > > oh, to use the text that appears when they're "beyond measurement." I could include sprites if needed, though I would have to change genitals.dm and do some math so they're not included in the player breast selection list.
> > 
> > 
> > Honestly more sprites and such would be based as fuck. Hyper has sprites if you need them ported but y'could always ask Arcie.
> 
> the style of hyper's breast sprites are different. i'll commit the new sprites soon + fix the PR title and description with previews, but i'm trying to stick with the current style in order for it not to feel out of place I'm also adding functionality for the ghetto gender swapping that was commented in but was never actually added

Fair enough! And I mean it doesnt technically have to be locked to just succubus milk but that'd be an interesting quirk to it. Might want to add a similar effect to incubus draft if thats the case but thats alot more work sprites wise.

**Comment by Gofawful5** [comment]
> > > > > Actually to be honest, I'm not sure why the cap needs to be increased anyway? There are already plenty of sizes.
> > > > 
> > > > 
> > > > oh, to use the text that appears when they're "beyond measurement." I could include sprites if needed, though I would have to change genitals.dm and do some math so they're not included in the player breast selection list.
> > > 
> > > 
> > > Honestly more sprites and such would be based as fuck. Hyper has sprites if you need them ported but y'could always ask Arcie.
> > 
> > 
> > the style of hyper's breast sprites are different. i'll commit the new sprites soon + fix the PR title and description with previews, but i'm trying to stick with the current style in order for it not to feel out of place I'm also adding functionality for the ghetto gender swapping that was commented in but was never actually added
> 
> Fair enough! And I mean it doesnt technically have to be locked to just succubus milk but that'd be an interesting quirk to it. Might want to add a similar effect to incubus draft if thats the case but thats alot more work sprites wise.

yeah, this is mainly just an overhaul intended for breasts

**Comment by RuneQuester** [comment]
This does not surprise me in the slightest that you would make this
Tracy "Big Chest" Abbenson

**Comment by linnpap** [comment]
what the fuck

**Comment by Gofawful5** [comment]
> what the fuck

yeah pretty much

**Comment by bluetshirtguy** [comment]
wow! extensive, now let's pray it's less buggy than sex2

**Comment by Gofawful5** [comment]
> wow! extensive, now let's pray it's less buggy than sex2

i've tested it rather extensively and i haven't had much (if any) issues with it so far on the latest commit
but if anything silly/stupid is found in code that could be improved i will get to it =)

**Comment by Gofawful5** [comment]
no decimals (they dont work ofc) so im rounding sizes

**Comment by optimumtact** [comment]
what are they putting in the water on skyrat station to make your spines so strong?

**Comment by tf-4** [comment]
ok that sprite is uh. A little excessive. I think the sizes were fine before

**Comment by gonenoculer5** [comment]
> what are they putting in the water on skyrat station to make your spines so strong?

All the phoron in the water is making our spines strong apparently.

**Comment by Gofawful5** [comment]
> ok that sprite is uh. A little excessive. I think the sizes were fine before

they're only obtainable via succubus milk (ie the breast expansion chem), i think it's only fair

**Comment by Mothblocks** [comment]
image

Bring her back

**Comment by Gofawful5** [comment]
> image
> 
> Bring her back

she got embarrassed

**Comment by tralezab** [comment]
Add her back king

**Comment by Fikou** [comment]
👎 bring it upstream

**Comment by Gofawful5** [comment]
> 👎 bring it upstream

u wish

**Comment by Mothblocks** [comment]
> she got embarrassed

thats fair enough

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (ChungusGamer666) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:308]: Please make these into static lists.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (ChungusGamer666) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:347]: Code will never get run due to the plentiful returns before this.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (ChungusGamer666) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:358]: Use an early return.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (ChungusGamer666) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:391]: Could be a single line with ||.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (ChungusGamer666) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:422]: ...what?

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (ChungusGamer666) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:438]: Please, make this a static list.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (ChungusGamer666) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:479]: Early return... Please.

**Review by ChungusGamer666 — CHANGES_REQUESTED** [review]
I didn't do the most thorough examination, but i did find plenty of problems.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: yeah that's fair

File: modular_skyrat/modules/customization/modules/surgery/organs/genitals.dm
Comment (Gofawful5) [reply modular_skyrat/modules/customization/modules/surgery/organs/genitals.dm]: that'sssss true actually, sure

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: oh no, i've tested that- it works just fine even with those returns.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:431]: where?

**Comment by tf-4** [comment]
Still entirely unconvinced by the sprites. Not sure an increase to the max size was necessary at all

**Comment by Gofawful5** [comment]
> Still entirely unconvinced by the sprites. Not sure an increase to the max size was necessary at all

Elaborate what you mean by being "unconvinced." And also of what you mean by it being necessary or not. Does it matter that much? Unnecessary does not equal bad, and this is a PR specifically to remedy certain aspects of the chemicals that make it more immersive, and allow for more ability within both scenes and RP.
Changing the max size wasn't _too_ necessary. But it doesn't cause any issues, and most I've heard are supportive of it. The change was mainly in order so that the absurd sizes text would be used, and it's still mechanically the same chem, just with added sizes over the max.
One of the main reasons I actually changed the max size was because, frankly, you could **get** to the max size in-game already, by simply selecting it in the character preference menu. I don't see the issue with it being like this if it's specifically the breast enlargement chemical.
I just think it'd be rather boring if you had the max size ingame, and then the milk did literally nothing to you. It's not immersive, nor is it intuitive.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (ChungusGamer666) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:364]: if(!exposed_mob.getorganslot(ORGAN_SLOT_BREASTS)
  return

Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:431]: but uhh why
i'll add it and test it real quick actually

Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm:431]: as expected, the code below that does not run
I don't understand the point of putting an early return there, since it's specifically for ODing on the breast enlargement chem, and is supposed to insert the breasts organ into the mob if they don't have breasts.

**Comment by gonenoculer5** [comment]
@Gofawful5 How we lookin?

**Comment by Gofawful5** [comment]
> @Gofawful5 How we lookin?

It's ready, just waiting on a review now.

**Comment by Gofawful5** [comment]
oh, I noticed one new problem with it- layering issues. i'll work on that now.

**Comment by Gofawful5** [comment]
ok fixed and now finally properly ready

**Comment by GoldenAlpharex** [comment]
I'm personally not convinced we really need even bigger genital sprites, one way or the other. The biggest ones we have right now are already feeling incredibly big as-is, I don't feel like this'll really help with that.

**Comment by Gofawful5** [comment]
> I'm personally not convinced we really need even bigger genital sprites, one way or the other. The biggest ones we have right now are already feeling incredibly big as-is, I don't feel like this'll really help with that.

really. so you're saying that if a character- with the current max breast size- drinks the Breast Enlargement Chemical and their boobs do not get bigger, are you telling me they should just RP it out without them actually increasing in size? i'm a bit confused with the logic here

can you explain to me why these sprites are unneeded, other than them being too big?

**Comment by gonenoculer5** [comment]
Probably should mark TF's requested changes as resolved btw if its done.

**Review by Gofawful5 — COMMENTED** [review]
this did nothing

**Comment by Gofawful5** [comment]
> Probably should mark TF's requested changes as resolved btw if its done.

issue. i cannot LOL
all i can do is see the review, not approve or re-request review

**Comment by tf-4** [comment]
I'm still not interested in the sprites, nor really the max size increase. They're too big. The current max size is fine IMO.

**Comment by Gofawful5** [comment]
> I'm still not interested in the sprites, nor really the max size increase. They're too big. The current max size is fine IMO.

okay, but can you give me a reason that isn't based on opinion?
Reiterating myself, the entire reason for the increase in size is so that succubus milk has an effect on people already at max size in character prefs. I've stated this before, and haven't heard anything by you or Golden that's a proper rebuttal. It makes sense for it to work, so, why _not?_

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (thestubborn) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]:
```suggestion
		var/static/list/cock_text_list = list("cock", "penis", "dick", "member", "richard", "johnston", "johnson")
```

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (thestubborn) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]:
```suggestion
		var/static/list/bigger_cock_text_list = list("rod", "shaft", "cock", "penis", "dick", "member", "richard", "johnston", "johnson",)
```
right and PROPER

Comment (thestubborn) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: also trailing commas, add commas to the end of all em :)

**Review by thestubborn — APPROVED** [review]

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm
Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: wait where does johnston come from

Comment (thestubborn) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: the short hand of richard is dick

Comment (Gofawful5) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm]: fair

**Comment by thestubborn** [comment]
awwwww

i'd try and talk to another maintainer, I do think that there should technically be sizes beyond the cap but


I'm not tom or golden

**Comment by Gofawful5** [comment]
> awwwww
> 
> i'd try and talk to another maintainer, I do think that there should technically be sizes beyond the cap but
> 
> I'm not tom or golden

apparently it was discussed among the maintainers, so /shrug

**Comment by tf-4** [comment]
> awwwww
> 
> i'd try and talk to another maintainer, I do think that there should technically be sizes beyond the cap but
> 
> I'm not tom or golden

The course of action you should take after a maintainer asks for changes should not be to go and find another one in the hopes that they approve it.

**Comment by Gofawful5** [comment]
hm

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**
